floss/language/go/extract.py

```diff
@@ -0,0 +1,511 @@
+    # 52 4D 54 72 77 73 7A 5A  39 57 7A 69 6C 64 2F 72  RMTrwszZ9Wzild/r
+    # 33 31 50 47 70 61 6B 2D  48 77 36 4B 72 77 59 6E  31PGpak-Hw6KrwYn
+    # 52 4E 73 22 0A 20 FF CC  CC CC CC CC CC CC CC CC  RNs". ..........
+
+    build_id_regex = re.compile(b"(?<=\xff\x20)(?P<buildid>(.)*)\x0A")
+
+    s = re.search(build_id_regex, section_data)
+    if s:
+        addr = s.start()
+        binary_string = s.group("buildid")
+
+        try:
+            string = [StaticString.from_utf8(binary_string, addr, min_length)]
+            return string
+        except ValueError:
+            pass
+
+    return []
+
+
+def extract_stackstring(extract_stackstring_pattern, section_data, min_length) -> List[StaticString]:
+    stack_strings = list()
+    for m in extract_stackstring_pattern.finditer(section_data):
+        for i in range(1, 8):
+            try:
+                binary_string = m.group(i)
+                if binary_string != b"":
+                    addr = m.start()
+                    try:
+                        string = StaticString.from_utf8(binary_string, addr, min_length)
+                        stack_strings.append(string)
+                    except ValueError:
+                        pass
+            except AttributeError:
+                break
+    return stack_strings
+
+
+def extract_reflection_strings(pe: pefile.PE, section_data, section_va, min_length) -> List[StaticString]:
+    # Extract string blob in .rdata section
+    # The data looks like <start_byte><length><string>  <start_byte><length><string>...
+    # <start_byte> is 0x00 or 0x01 or 0x02 or 0x03
+
+    # Data:-
+    """
+    0048E620  5B 34 5D 75 69 6E 74 38  00 09 2A 5B 38 5D 69 6E  [4]uint8..*[8]in
+    0048E630  74 33 32 00 09 2A 5B 38  5D 75 69 6E 74 38 00 09  t32..*[8]uint8..
+    0048E640  2A 5B 5D 73 74 72 69 6E  67 00 09 2A 5B 5D 75 69  *[]string..*[]ui
+    0048E650  6E 74 31 36 00 09 2A 5B  5D 75 69 6E 74 33 32 00  nt16..*[]uint32.
+    0048E660  09 2A 5B 5D 75 69 6E 74  36 34 00 09 2A 63 68 61  .*[]uint64..*cha
+    0048E670  6E 20 69 6E 74 01 09 41  6E 6F 6E 79 6D 6F 75 73  n int..Anonymous
+    0048E680  01 09 43 61 6C 6C 53 6C  69 63 65 01 09 43 6C 65  ..CallSlice..Cle
+    0048E690  61 72 42 75 66 73 01 09  43 6F 6E 6E 65 63 74 45  arBufs..ConnectE
+    0048E6A0  78 01 09 46 74 72 75 6E  63 61 74 65 01 09 49 6E  x..Ftruncate..In
+    0048E6B0  74 65 72 66 61 63 65 01  09 4E 75 6D 4D 65 74 68  terface..NumMeth
+    0048E6C0  6F 64 01 09 50 72 65 63  69 73 69 6F 6E 01 09 52  od..Precision..R
+    """
+
+    # The blob start like 00 00 <data> and ends like this:
+    """
+    0000000000492A40  6C 65 63 74 2E 53 74 72  75 63 74 46 69 65 6C 64  lect.StructField
+    0000000000492A50  2C 20 62 6F 6F 6C 29 00  00 00 00 00 00 00 00 00  , bool).........
+    0000000000492A60  08 00 00 00 00 00 00 00  08 00 00 00 00 00 00 00  ................
+    """
+
+    # Get the blob
+    blob_end_pattern = re.compile(b"\x00{8}.\x00{7}.\x00{7}")
```

Comment:
why does the data end like this? is it guaranteed to always look like this?

is this what you've seen in one program? many programs?

i think if we want to use a pattern like this, we should be able to prove why it works. a link to go compiler source or an explanation for why this pattern is used would help a lot.